.github/workflows/welcome-new-contributors.yaml

```diff
@@ -0,0 +1,26 @@
+name: 'Welcome New Contributors'
+
+on:
+  issues:
+    types: [opened]
+  pull_request_target:
```

Comment:
Sorry @pnbrown for coming back with one more late comment, but I noticed `pull_request_target` which is suggested in https://github.com/garg3133/welcome-new-contributors but it is discouraged in context of `GITHUB_TOKEN` in various places: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target and https://securitylab.github.com/research/github-actions-preventing-pwn-requests/.  I'm not too familiar with the details (still reading) and it might not apply here since there is no checkout in this workflow, but just a heads up.

Reply:
From what I'm reading, this use seems safe to me since the workflow does not run untrusted code (other than the action itself). However this is the first use of `pull_request_target` in this repo. The other similar action ([first-interaction](https://github.com/marketplace/actions/first-interaction)) seems to suggest `pull_request` instead of `pull_request_target`, see example [here](https://github.com/actions/first-interaction/blob/main/.github/workflows/first-interaction.yml). 

The problem with `pull_request_target` seems to be that`GITHUB_TOKEN` will have _write access_ to the repo, unlike the `GITHUB_TOKEN` which is issued in case of `pull_request`. I'm unsure if `pull_request_target` is really needed in order to comment on an issue and pull request? I guess the use of `pull_request_target` could be just mistake on [welcome-new-contributors](https://github.com/garg3133/welcome-new-contributors) docs.

Reply:
It looks like the other uses just pull_request

https://github.com/actions/first-interaction/blob/main/.github/workflows/first-interaction.yml

Reply:
Nice 👍   Do we have someone to create their first issue/pull request, to check that the action will still work? 😄

Reply:
I'm asking around 😅

Reply:
@tsaarni thanks for digging into this, I've been reading through the links you provided. From my read (particularly based on https://securitylab.github.com/research/github-actions-preventing-pwn-requests/), it seems like we will need to use `pull_request_target` to allow comments on PRs opened from forks, but I'm also happy to try it using `pull_request` first to see if it does work since the docs aren't 100% clear.

If we do need to end up switching back to `pull_request_target`, I agree that this usage seems safe since it's not executing any untrusted code. We could also use https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions to pare down the write permissions that are granted to just issues and PRs.